core/schema/container.go

```diff
@@ -475,7 +475,7 @@ func (s *containerSchema) Install() {
 			absolutely necessary and only with trusted commands.`),
 
 		dagql.NodeFunc("terminal", s.terminal).
-			Doc(`Return an interactive terminal for this container using its configured default terminal command if not overridden by args (or sh as a fallback default).`).
+			Doc(`Opens an interactive terminal for this container using its configured default terminal command if not overridden by args (or sh as a fallback default).`).
```

Comment:
One thing I just remembered, we want to call `.Impure(...)` on these because they are side-effectful now rather than content defined. Otherwise dagql caching will prevent the same call from running more than once in a session (which isn't gonna be a common case, but still realistic).

Same for the API on directory.

Reply:
Done!

I wasn't aware of `.Impure()` ... does this mean it would be technically possible to return the modified container/directory without affecting cache?

Reply:
No `Impure` only affects dagql-level caching, buildkit caching is completely separate atm.
* (tangential) If we move off LLB then things like this would be possible, just noting since this topic keeps popping it's head up in random places a lot recently...

Reply:
Then disregard the second half of my comment below :)